### PR TITLE
Much, much faster extract() implementation

### PR DESCRIPTION
**Pull request by simonw**
Takes my test down from ten minutes to four seconds. Refs #172.

### CODE REVIEW

**Comment by simonw** [comment]
This could do with a little bit more testing - I'm worried there may be column or table name edge cases that are not covered yet. I also need to remove the progress bar code since that no longer makes sense for this implementation.

**Comment by simonw** [comment]
After running through the steps in https://simonwillison.net/2020/Sep/23/sqlite-utils-extract/ I get a table that looks like this:

salaries__salaries__683_558_rows

The foreign key columns are all at the end of the table. It would be nicer if they were arranged in the same order as the columns they replaced.

**Comment by simonw** [comment]
`import functools` is no longer needed.

**Comment by simonw** [comment]
I could also be a bit smarter about transaction handling. I think it may be possible to run this entire operation in a single transaction now.

**Comment by simonw** [comment]
The way the lookup table works here differs from the previous implementation. In the previous implementation the usage of `.lookup()` meant that an existing table would be modified to fit the new purpose. That no longer happens in this version. Need to make a design decision about how this should work.

It should definitely be possible to use an existing lookup table - imagine a database where several tables have a "Departments" column and we want to extract all of those values out to a single shared "Departments" table.

**Comment by simonw** [comment]
I may revert the now unnecessary undocumented tweaks to the `.update()` method made in 66d506587eba9f0715267d6560b97c1fa44cc781 as well.

**Comment by simonw** [comment]
For reusing the lookup table: I'm going to raise an error if a lookup table exists but without the correct columns. The caller can then add those columns and try again.

File: sqlite_utils/db.py [hunk 1]
```diff
@@ -886,38 +892,85 @@ def extract(self, columns, table=None, fk_column=None, rename=None, progress=Non
...
+        # Populate the lookup table with all of the extracted unique values
+        lookup_columns_definition = {
+            (rename.get(col) or col): typ
+            for col, typ in self.columns_dict.items()
+            if col in columns
+        }
+        if lookup_table.exists() and not set(lookup_columns_definition.keys()).issubset(
+            lookup_table.columns_dict.keys()
+        ):
+            # TODO: Write test for this
```
Comment (simonw) [inline sqlite_utils/db.py:907]: Still need to write this test.

File: sqlite_utils/db.py [hunk 2]
```diff
@@ -731,6 +730,7 @@ def transform(
         not_null=None,
         defaults=None,
         drop_foreign_keys=None,
+        column_order=None,
```
Comment (simonw) [inline sqlite_utils/db.py:733]: Need to update docs for `.transform()` now that `column_order=` is available.

File: sqlite_utils/db.py (hunk 1 above)
Comment (simonw) [reply sqlite_utils/db.py:907]: The tests should also cover the case of reusing a lookup table.
